src/deadline/job_attachments/caches/cache_db.py

```diff
@@ -63,24 +67,37 @@ def __enter__(self):
         if self.enabled:
             import sqlite3
 
+            @_retry(
+                ExceptionToCheck=sqlite3.OperationalError,
+                tries=self._RETRY_ATTEMPTS,
+                delay=(0.5, 1.5),  # Jitter between 0.5 and 1.5 seconds
+                backoff=1.0,
+                logger=logger.warning,
+            )
+            def _connect_to_db():
+                try:
+                    connection = sqlite3.connect(self.cache_dir, check_same_thread=False)
+                    connection.execute("PRAGMA journal_mode=WAL")
+                    try:
+                        # Test the connection by trying to query the table
+                        connection.execute(f"SELECT * FROM {self.table_name}")
+                    except Exception:
+                        # DB file doesn't have our table, so we need to create it
+                        logger.info(
+                            f"No cache entries for the current library version were found. Creating a new cache for {self.cache_name}"
+                        )
+                        connection.execute(self.create_query)
+                    return connection
+                except sqlite3.OperationalError as oe:
+                    logger.info("Error connecting to database, retrying.")
```

Comment:
Can this be part of the `_retry` instead of duplicating it here? Looks like this will print that it's retrying even for the cycle where it doesn't.